packages/flutter/test/material/banner_test.dart

```diff
@@ -1105,6 +1105,27 @@ void main() {
       ),
     );
   });
+
+   testWidgets('Custom Margin respected', (WidgetTester tester) async {
+    const EdgeInsets margin = EdgeInsets.all(30);
+    await tester.pumpWidget(
+      MaterialApp(
+        home: MaterialBanner(
+         margin: margin,
+          content: const Text('I am a banner'),
+          actions: <Widget>[
+            TextButton(
+              child: const Text('Action'),
+              onPressed: () { },
+            ),
+          ],
+        ),
+      ),
+    );
+
+  final Container materialContainer = tester.widget<Container>(find.descendant(of: find.byType(MaterialBanner), matching: find.byType(Container)).first);
```

Comment:
This is OK but makes the test unnecessarily depend on an implementation detail (that margin is implemented with a Container). It would be better to verify that the bounds of the banner are what's expected when the banner is displayed.

Reply:
I have used the same approach that other tests are using in `banner_test.dart`
I have written a test which finds the material and then gets the bounds of the material.